applications/StructuralMechanicsApplication/custom_elements/timoshenko_curved_beam_element_2D3N.h

```diff
@@ -0,0 +1,492 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:  Alejandro Cornejo
+//
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/element.h"
+#include "custom_utilities/structural_mechanics_element_utilities.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class LinearTimoshenkoCurvedBeamElement2D3N
+ * @ingroup StructuralMechanicsApplication
+ * @brief This is the Timoshenko curved beam element of 3 nodes. Reference:
+ * Connecting beams and continua: variational basis and mathematical analysis, Romero and Schenk, Meccanica, 2023
+ * DOI: https://doi.org/10.1007/s11012-023-01702-0
+ * 
+ * Ordering of the nodes:      0 ------ 2 ------- 1
+ * 
+ * Quadratic interpolation of the curved geometry, displacements and rotation. Reduced integration to avoid shear locking
+ * @author Alejandro Cornejo
+ */
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) LinearTimoshenkoCurvedBeamElement2D3N
```

Comment:
I find the name a bit misleading. Some questions/comments:
- Does it make sense to have the straight version? Is it a big difference in computational effort?
- I would rename this element to `LinearTimoshenkoBeamElement2D3N` and the other one `LinearTimoshenkoStraightBeamElement2D3N` if you want to keep both, I think this makes misuse harder
- Do you have checks in the straight element to prevent curved geometries?